src/mono/mono/mini/mini-llvm.c

```diff
@@ -8433,7 +8433,21 @@ MONO_RESTORE_WARNING
 			LLVMValueRef dest;
 
 			dest = convert (ctx, LLVMBuildAdd (builder, convert (ctx, values [ins->inst_destbasereg], IntPtrType ()), LLVMConstInt (IntPtrType (), ins->inst_offset, FALSE), ""), pointer_type (t));
-			mono_llvm_build_aligned_store (builder, lhs, dest, FALSE, 1);
+			if (mono_class_value_size (ins->klass, NULL) == 12) {
+				const int mask_values [] = { 0, 1, 2 };
+
+				LLVMValueRef truncatedVec3 = LLVMBuildShuffleVector (
+					builder,
+					lhs,
+					LLVMGetUndef (t),
+					create_const_vector_i32 (mask_values, 3),
+					"truncated_vec3"
+				);
+
+				mono_llvm_build_aligned_store (builder, truncatedVec3, dest, FALSE, 1);
```

Comment:
Why is the alignment being set to `1`? A `Vector3` is composed of `3x float` fields and so is presumed to be 4 byte aligned unless the code is explicitly using something like `Unsafe.ReadUnaligned` or `Unsafe.WriteUnaligned` (which use the IL `unaligned. ` prefix).

Reply:
This change reuses existing logic. This code path was previously taken for all `VectorX` writes so I kept it consistent. I agree that this might be incorrect. We should investigate this separately.